contracts/contracts/strategies/Generalized4626Strategy.sol

```diff
@@ -0,0 +1,141 @@
+     * @param _amount Amount of asset to deposit
+     */
+    function _deposit(address _asset, uint256 _amount) internal {
+        require(_amount > 0, "Must deposit something");
+        require(_asset == address(assetToken), "Unexpected asset address");
+
+        IERC4626(platformAddress).deposit(_amount, address(this));
+        emit Deposit(_asset, address(shareToken), _amount);
+    }
+
+    /**
+     * @dev Deposit the entire balance of assetToken to gain shareToken
+     */
+    function depositAll() external override onlyVault nonReentrant {
+        uint256 balance = assetToken.balanceOf(address(this));
+        if (balance > 0) {
+            _deposit(address(assetToken), balance);
+        }
+    }
+
+    /**
+     * @dev Withdraw asset by burning shares
+     * @param _recipient Address to receive withdrawn asset
+     * @param _asset Address of asset to withdraw
+     * @param _amount Amount of asset to withdraw
+     */
+    function withdraw(
+        address _recipient,
+        address _asset,
+        uint256 _amount
+    ) external override onlyVault nonReentrant {
+        require(_amount > 0, "Must withdraw something");
+        require(_recipient != address(0), "Must specify recipient");
+        require(_asset == address(assetToken), "Unexpected asset address");
+
+        IERC4626(platformAddress).withdraw(_amount, _recipient, address(this));
+        emit Withdrawal(_asset, address(shareToken), _amount);
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset / share tokens
+     * @param _asset Address of the asset to approve
+     * @param _pToken The pToken for the approval
+     */
+    function _abstractSetPToken(address _asset, address _pToken)
+        internal
+        override
+    {
+        shareToken = IERC20(_pToken);
+        assetToken = IERC20(_asset);
+
+        // Safe approval
+        shareToken.safeApprove(platformAddress, type(uint256).max);
+        assetToken.safeApprove(platformAddress, type(uint256).max);
+    }
+
+    /**
+     * @dev Remove all assets from platform and send them to Vault contract.
+     */
+    function withdrawAll() external override onlyVaultOrGovernor nonReentrant {
+        uint256 shareBalance = shareToken.balanceOf(address(this));
+        uint256 assetAmount = IERC4626(platformAddress).redeem(
+            shareBalance,
+            vaultAddress,
+            address(this)
+        );
+        emit Withdrawal(address(assetToken), address(shareToken), assetAmount);
+    }
+
+    /**
+     * @dev Get the total asset value held in the platform
+     * @param _asset      Address of the asset
+     * @return balance    Total value of the asset in the platform
+     */
+    function checkBalance(address _asset)
+        external
+        view
+        override
+        returns (uint256 balance)
+    {
+        require(_asset == address(assetToken), "Unexpected asset address");
+        return IERC4626(platformAddress).totalAssets();
```

Comment:
@sparrowDom, This seems wrong. TotalAssets isn't how many assets we are owed by the downstream vault, but how many assets it contains in total, for all users.

Reply:
Yup this was very wrong replaced with: 
```
return
            IERC4626(platformAddress).convertToAssets(
                shareToken.balanceOf(address(this))
            );
```